tinygrad/tensor.py

```diff
@@ -153,12 +153,39 @@ def empty(*shape, **kwargs):
     return Tensor._loadop(LoadOps.EMPTY, prod((shape:=argfix(*shape))), **kwargs).reshape(shape)
 
   _seed: int = int(time.time())
+  _rng_counter: int = 0 # TODO: this has to be per device
   @staticmethod
   def manual_seed(seed=0): Tensor._seed = seed
 
+  @staticmethod
+  def _rand(): # threefry algo
+    key = [Tensor._seed, 0x0] # TODO: replace 0x0 with the CPU/GPU device this is running on 
```

Comment:
the `0x0` should be the device the tensor is running on. from my reading, it doesn't really matter which of the 4 values of key/x you change, but if all 4 are the same you get the same output